dipy/io/stateful_tractogram.py

```diff
@@ -548,8 +548,11 @@ def remove_invalid_streamlines(self, epsilon=1e-3):
             indices_to_remove.append(bisect(
                 self._tractogram.streamlines._offsets, i) - 1)
 
-        indices_to_keep = np.setdiff1d(np.arange(len(self._tractogram)),
-                                       np.array(indices_to_remove)).astype(int)
+        indices_to_remove = sorted(set(indices_to_remove))
```

Comment:
I do not understand why you need to sort them. Is it just for the test? Can you explain?

I suppose, the use of `set` is due to multiple duplications, right?

Reply:
ok, I see for the `set` now.

Reply:
Please read the commit message in 07bf4c2. Although sorting is not strictly necessary, on the one hand it makes sense to return sorted values and on the other hand it helps when you need to explore the values. Also, the indices to keep are sorted by construction.